src/main/java/seedu/address/model/person/SchoolResidence.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class SchoolResidence {
+
+    private static final List<String> listResidences = Arrays.asList("PGPH", "PGPR", "KE7H", "SH", "KRH", "TH", "EH",
+            "RH", "RVRC", "YNC", "TC", "CAPT", "RC4", "USP", "UTR");
+
+    public static final String MESSAGE_CONSTRAINTS = "The residence entered should be one of the following: \n"
+            + listResidences.toString();
+
+    public final String value;
```

Comment:
Remember to capitalize the name for your constant variables to adhere to Java coding standard.